resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverContext.java

```diff
@@ -135,11 +135,18 @@ public void operationComplete(Future<T> future) throws Exception {
                     }
                 }
             });
-        }
-        if (parent.searchDomains(hostname)) {
+            int idx = hostname.length();
+            int dots = 0;
+            while (idx-- > 0) {
+                if (hostname.charAt(idx) == '.') {
+                    dots++;
```

Comment:
nit: consider combining this with the previous `if` statement to potentially limit branch instructions

```java
if (hostname.charAt(idx) == '.' && ++dots >= parent.ndoes()) {

}
```